libs/shared/selectors/src/selectors/flags-selector/flags.service.ts

```diff
@@ -7,7 +7,9 @@ import { Flag } from '@keira/shared/constants';
 export class FlagsService {
   getBitsFromValue(value: number, bitsCount: number): boolean[] {
     const bits = new Array<boolean>(bitsCount);
-    const binaryStr: string = value.toString(2).split('').reverse().join('');
+
+    let binaryStr: string = '0'.repeat(bitsCount);
+    if (value !== -1) binaryStr = value.toString(2).split('').reverse().join('');
```

Comment:
does this 1-line work too?

```suggestion
    const binaryStr: string = value !== -1 ? value.toString(2).split('').reverse().join('') : '0'.repeat(bitsCount);
```